pandas/core/series.py

```diff
@@ -1143,9 +1143,9 @@ def __setitem__(self, key, value) -> None:
 
     def _set_with_engine(self, key, value) -> None:
         loc = self.index.get_loc(key)
-        # error: Argument 1 to "validate_numeric_casting" has incompatible type
-        # "Union[dtype, ExtensionDtype]"; expected "dtype"
-        validate_numeric_casting(self.dtype, value)  # type: ignore[arg-type]
+        if not can_hold_element(self._values, value):
```

Comment:
i assume these are caught elsewhere otherwise you'd have a message here right?

Reply:
correct